lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -77,41 +77,22 @@ def _provide_loki(self):
                 self.unit.status = BlockedStatus(str(e))
 
 
-The `LokiPushApiProvider` object has two main responsibilities:
+The `LokiPushApiProvider` object has several main responsibilities:
 
-1.- Be in charge of setting the Loki Push API Address into
-relation data so clients can use it to send logs. Every time a unit joins
-a relation: `$ juju add-relation loki-k8s loki-client-k8s` the object sets:
+1. Set the URL of the Loki Push API in the provider app data bag; the URL
+   must be unique to all instances (e.g., using a load balancer).
```

Comment:
This has a data loss potential during the time that no leader is present, e.g. during scale down.
BTW, I did not see any leader events registered.
I do recall this is intended as a single unit, so might worth noting that assumption.
Also, not sure what it means that no peer relation is defined in metadata.

Reply:
I don't understand this:

> the URL must be unique to all instances (e.g., using a load balancer).

Reply:
> This has a data loss potential during the time that no leader is present, e.g. during scale down.
BTW, I did not see any leader events registered.
I do recall this is intended as a single unit, so might worth noting that assumption.
Also, not sure what it means that no peer relation is defined in metadata.

The API will need to work transparently with Loki charmed in distributed mode, so better to bite that bullet now.

Reply:
> Also, not sure what it means that no peer relation is defined in metadata.

It means nothing special. We could use a peer relation to observe wether the Loki charm is scaled to more than one unit, and put it in blocked state like we do it with Prometheus + Ingress thou.